pkg/controller/cassandradatacenter/statefulset.go

```diff
@@ -91,10 +92,11 @@ func newStatefulSetSpec(cdc *cassandraoperatorv1alpha1.CassandraDataCenter, podS
 func newPodSpec(cdc *cassandraoperatorv1alpha1.CassandraDataCenter, volumes []corev1.Volume, containers []corev1.Container, initContainers []corev1.Container) *corev1.PodSpec {
 	// TODO: should this spec be fully exposed into the CDC.Spec?
 	podSpec := &corev1.PodSpec{
-		Volumes:          volumes,
-		Containers:       containers,
-		InitContainers:   initContainers,
-		ImagePullSecrets: cdc.Spec.ImagePullSecrets,
+		Volumes:            volumes,
+		Containers:         containers,
+		InitContainers:     initContainers,
+		ImagePullSecrets:   cdc.Spec.ImagePullSecrets,
+		ServiceAccountName: cdc.Spec.ServiceAccountName,
```

Comment:
What happens if ServiceAccountName is not set in the CRD? Will this work with empty string?

Reply:
Yes, we are using `omitempty` in the spec so omitting this field simply creates the StatefulSet without the `ServiceAccountName` field. It's the same as the existing `ImagePullSecrets` field.

https://github.com/instaclustr/cassandra-operator/pull/216/files#diff-ef654dce75ffebdf8a5ce52918b76039R27